docs/user/metrics/custom_distribution.md

```diff
@@ -0,0 +1,86 @@
+# Custom Distribution
+
+Custom distributions are used to record the distribution of arbitrary values.
+
+It should be used only when direct control over how the histogram buckets are computed is required.
+Otherwise, look at the standard distribution metric types:
+
+* [Timing Distributions](timing_distribution.md))
+
+> Note: Custom distributions are currently only allowed for GeckoView metrics (the `gecko_datapoint` parameter is present).
+
+## Configuration
+
+Custom distributions have the following required parameters:
+
+  - `range_min`: The minimum value of the first bucket
+  - `range_max`: The minimum value of the last bucket
+  - `bucket_count`: The number of buckets
+  - `histogram_type`: 
+    - `linear`: The buckets are evenly spaced
+    - `exponential`: The buckets follow a natural logarithmic distribution
+
+In addition, the metric should specify:
+
+  - `unit`: The unit of the values in the metric. For documentation purposes only -- does not affect data collection.
+
+If you wanted to create a custom distribution to the fraction of memory that is overhead as a percentage, first you need to add an entry for it to the `metrics.yaml` file:
+
+```YAML
+memory:
+  heap_overhead_fraction:
+    type: custom_distribution
+    description: >
+      Fraction of committed heap memory that is overhead (percentage).
+    range_min: 1
+    range_max: 100
+    bucket_count: 25
+    histogram_type: linear
+    unit: percentage
+    gecko_datapoint: MEMORY_HEAP_OVERHEAD_FRACTION
+    ...
+```
+
+## API
+
+Now you can use the custom distribution from the application's code.
+
+```Kotlin
+import org.mozilla.yourApplication.GleanMetrics.Memory
+
+Memory.heapOverheadFraction.accumulateSamples([23])
```

Comment:
It doesn't have a method to just accumulate a single value?

Reply:
Not at present, but that's basically an artifact of the fact that this is currently GeckoView-only.  @Dexterp37: What do you think?  Should we add a single sample method (even though nothing will use it?)

Reply:
I don't think we should since, as you stated, nothing will really be using it :)